flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/FactoryUtil.java

```diff
@@ -378,13 +376,17 @@ private TableFactoryHelper(DynamicTableFactory tableFactory, DynamicTableFactory
 		/**
 		 * Discovers a {@link ScanFormat} of the given type using the given option as factory identifier.
 		 *
-		 * <p>A prefix, e.g. {@link #KEY_FORMAT_PREFIX}, projects the options for the format factory.
+		 * <p>The format option key must be 'format' or with '.format' suffix. The discovery logic
+		 * will replace 'format' with the factory identifier value as the format prefix.
+		 * For example, assuming the identifier is 'json', if format option key is 'format',
+		 * then format prefix is 'json.'. If format option key is 'value.format', then format
+		 * prefix is 'value.json'. The format prefix is used to project the options for the
+		 * format factory.
 		 */
 		public <I, F extends ScanFormatFactory<I>> ScanFormat<I> discoverScanFormat(
 				Class<F> formatFactoryClass,
-				ConfigOption<String> formatOption,
-				String formatPrefix) {
-			return discoverOptionalScanFormat(formatFactoryClass, formatOption, formatPrefix)
+				ConfigOption<String> formatOption) {
```

Comment:
If we require an `formatOption` here, we should definitely provide the default ones as static fields in `FactoryUtil`. We can then also link them in the Java doc. Instead of copying this big text 4 times, I would suggest to add it to the `createTableFactoryHelper`.